src/components/signUpForm/SignUpForm.js

```diff
@@ -1,38 +1,74 @@
-import React from 'react';
+import React, { useState } from 'react';
 import { NavLink } from 'react-router-dom';
+import { useDispatch, useSelector } from 'react-redux';
 
 import './SignUpForm.css';
+import * as userActions from '../../redux/actions/userActions';
 
-const SignUpForm = () => (
-  <>
-    <form className="form">
-      <div className="form_input">
-        <span className="form_input__title">email</span>
-        <input type="text" className="form_input__field" />
-      </div>
-      <div className="form_input">
-        <span className="form_input__title">password</span>
-        <input type="password" className="form_input__field" />
-      </div>
-      <div className="form_input">
-        <input
-          type="submit"
-          className="form_input__field form_input__field--color"
-          value="sign in"
-        />
-      </div>
-    </form>
-    <NavLink to="/forgot-password" className="column__forgot-password">
-      Forgot your password?
-    </NavLink>
-    <NavLink to="/facebook-login" className="column__conect-with-facebook">
-      connect with facebook
-    </NavLink>
-    <div className="column__line"></div>
-    <NavLink to="/sign-up" className="column__sign-up">
-      sign up
-    </NavLink>
-  </>
-);
+const SignUpForm = () => {
+  const [user, setUser] = useState({ email: '', password: '' });
+  const error = useSelector(state => state.user.error);
+  const dispatch = useDispatch();
+
+  const handleEmailChange = event => {
+    const newUser = { ...user, email: event.target.value };
+    setUser(newUser);
+  };
+
+  const handlePasswordChange = event => {
+    const newUser = { ...user, password: event.target.value };
+    setUser(newUser);
+  };
```

Comment:
Not saying this is the better way, just an idea.

```suggestion
  const handleFieldChange = ({ target: { value } }, key) => 
    setUser({ ...user, [key]: value });
```